docs/source/config/index.rst

```diff
@@ -0,0 +1,292 @@
+      id: beamline_setup
+      components:
+        - label: <UI label>
+          attribute: <beamline_attribute>
+          step: <step_size>
+          precision: <value_precision>
+          suffix: <value_suffix>
+
+The ``components`` key contains a list of control widget specifications.
+Each widget specification have the following keys:
+
++-----------+-------------------------------------+
+| key       | purpose                             |
++===========+=====================================+
+| attribute | hardware beamline attribute id      |
++-----------+-------------------------------------+
+| label     | widget's text label                 |
++-----------+-------------------------------------+
+| step      | widget's single step size           |
++-----------+-------------------------------------+
+| precision | widget's position display precision |
++-----------+-------------------------------------+
+| suffix    | widget's position units suffix      |
++-----------+-------------------------------------+
+
+Note that ``step``, ``precision`` and ``suffix`` key are only used by continuous widgets and are omitted for discrete widgets.
+
+Here is a short example of ``beamline_setup`` section:
+
+.. code-block:: yaml
+
+    beamline_setup:
+      id: beamline_setup
+      components:
+        - label: Fast Shutter
+          attribute: fast_shutter
+        - label: Safety shutter
+          attribute: safety_shutter
+        - label: Energy
+          attribute: energy
+          step: 0.001
+          precision: 4
+          suffix: keV
+        - label: Resolution
+          attribute: resolution
+          step: 0.001
+          precision: 3
+          suffix: Å
+
+camera_setup
+------------
+
+Configures video stream widgets.
+Each widget provides access to a video stream on the *Data collection* tab.
+This is typically used for giving users an easy way to access hutch view cameras.
+The section have the following syntax:
+
+.. code-block:: yaml
+
+    camera_setup:
+      id: camera_setup
+      components:
+        - label: <UI label>
+          attribute: <stream format>
+          url: <video stream url>
+          width: <width>
+          height: <height>
+
+The ``components`` key contains a list of video stream specifications.
+Each stream specification have the following keys:
+
++-----------+---------------------+
+| key       | purpose             |
++===========+=====================+
+| label     | widget's text label |
++-----------+---------------------+
+| attribute | stream format       |
++-----------+---------------------+
+| url       | video stream's URL  |
++-----------+---------------------+
+| width     | video window width  |
++-----------+---------------------+
+| height    | video window height |
++-----------+---------------------+
+
+The ``attribute`` specifies the video format.
+If ``attribute`` is set to 'mp4', the video will be rendered with ``<video>`` HTML tag, otherwise ``<img>`` tag is used.
```

Comment:
```suggestion
If ``attribute`` is set to ``mp4``, the video will be rendered with the ``<video>`` HTML tag, otherwise the ``<img>`` tag is used.
```